test/with_dummyserver/test_poolmanager.py

```diff
@@ -614,18 +614,19 @@ def test_top_level_request_with_timeout(self) -> None:
         ],
     )
     def test_request_with_json(self, headers: HTTPHeaderDict) -> None:
+        old_headers = None if headers is None else headers.copy()
         body = {"attribute": "value"}
         r = request(
             method="POST", url=f"{self.base_url}/echo_json", headers=headers, json=body
         )
         assert r.status == 200
         assert r.json() == body
-        if headers is not None and "application/json" not in headers.values():
-            assert "text/plain" in r.headers["Content-Type"].replace(" ", "").split(",")
-        else:
-            assert "application/json" in r.headers["Content-Type"].replace(
-                " ", ""
-            ).split(",")
+        content_type = None if headers is None else headers.get("content-Type")
+        content_type = "application/json" if content_type is None else content_type
```

Comment:
These lines are a bit hard to read. What do you think about changing this to one of the following variants?

```suggestion
        if not headers:
            content_type = "application/json"
        else:
            content_type = headers.get("content-Type", "application/json")
```
```suggestion
        content_type = (headers or {}).get("content-Type", "application/json")
```

Reply:
if you really mean "what do you think?": then I find the way that I wrote it more readable.

if you really mean "please change it" then sure, I dont care: please say that!

Reply:
I changed these lines in #3203, thanks for taking the bugfix into a separate PR!